### PR TITLE
Fix index set defaults change not being reflected in form

### PR DESCRIPTION
**Pull request by grotlue**
Fixes https://github.com/Graylog2/graylog2-server/issues/18049

## Description
- This PR enables reinitialization in the index set formik form, so it can be updated with new defaults. 
- It also now fetches the defaults in the form component directly instead of passing it through from the creation page.
This is necessary due to the new Data Tiering feature as the legacy tab needs to show the correct default values when editing an index set with Data Tiering configured.
- While on it, I also fixed a warning about an unused prop in `<RangeInput/>`

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-web-interface/src/components/indices/IndexSetConfigurationForm.tsx [hunk 1]
```diff
@@ -186,9 +186,35 @@ const IndexSetConfigurationForm = ({
...
   const history = useHistory();
-  const [indexSetState] = useState<IndexSet>(indexSet);
+  const [indexSet, setIndexSet] = useState(initialIndexSet);
+
   const [fieldTypeRefreshIntervalUnit, setFieldTypeRefreshIntervalUnit] = useState<Unit>('seconds');
   const { loadingIndexDefaultsConfig, indexDefaultsConfig } = useIndexDefaults();
+
+  useEffect(() => {
+    if (loadingIndexDefaultsConfig || !indexDefaultsConfig) return;
+
```
Comment (gally47) [inline graylog2-web-interface/src/components/indices/IndexSetConfigurationForm.tsx:196]: I think here it makes sense to create a hook useIndexSet that contains all the useEffect logic + useState, to keep the component light and readable it already has a lot of lines and functions, what do you think?

**Review by gally47 — CHANGES_REQUESTED** [review]
Tested it, issue fixed, code LGTM!
I have a small suggestion:

File: graylog2-web-interface/src/components/indices/IndexSetConfigurationForm.tsx (hunk 1 above)
Comment (grotlue) [reply graylog2-web-interface/src/components/indices/IndexSetConfigurationForm.tsx]: Good idea, did that. Can you test it again?

**Review by gally47 — APPROVED** [review]
LGTM!

**Pull request merged**